rust/noosphere/src/platform.rs

```diff
@@ -3,12 +3,11 @@
 ///! secure key management. This module lays out the concrete strategies we will
 ///! use on a per-platform basis.
 
-#[cfg(all(
-    any(target_arch = "aarch64", target_arch = "x86_64"),
-    target_vendor = "apple"
-))]
+#[cfg(all(feature = "rocksdb", feature = "sqlite",))]
+compile_error!("Cannot enable multiple storage backends.");
```

Comment:
FWIW Rust suggested practice is to gracefully accept features that ought to be mutually exclusive: https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features

Reply:
Also: if we do go down the suggested route here, we need to include all the back-ends which today includes `sled` and `indexed_db`.

You might get better mileage by using conditional selection in the `noosphere/build.rs` crate you added so that native backends have a well-defined order of precedence (e.g., if `all(rocks, sled)` then rocks wins, and subsequently if `all(rocks, sqlite)` the SQLite wins; `wasm` results in any native backend being disregarded etc.).

Reply:
SG, with `cfg_aliases` this becomes easy to have ordinal usage e.g. `--features rocksdb --features sqlite` would prefer e.g. rocksdb